src/publisher/api_v2_views.py

```diff
@@ -107,16 +110,18 @@ def article_version(request, id, version):
 
 @api_view(['POST'])
 def article_fragment(request, art_id, fragment_id):
-    only_published = is_authenticated(request)
+    if not is_authenticated(request):
+        return Response("not authenticated. only authenticated admin users can modify content", status=403)
     try:
         reserved_keys = [XML2JSON]
         ensure(fragment_id not in reserved_keys, "that key is taken")
         with transaction.atomic():
-            av = logic.most_recent_article_version(art_id, only_published)
-            data = request.data
-            frag, created, updated = fragment_logic.add(av.article, fragment_id, data, update=True)
-            ensure(created or updated, "fragment was not created/updated")
-            fragment_logic.merge_if_valid(av, quiet=False)
+            avs = logic.article_version_list(art_id, only_published=False)
+            for av in avs:
```

Comment:
if you create a fragment, attach it to an article and don't specify a version, it gets merged for *all* article versions.

when the `xml->json` fragment is created, it's attached to a specific article version.

this way we can have article-global overrides and articleversion-specific overrides using fragments.

if it's working for now, that's fine, but it is on my radar to fix.